test/es-module/test-esm-loader-hooks.mjs

```diff
@@ -422,18 +423,118 @@ describe('Loader hooks', { concurrency: true }, () => {
     });
   });
 
-  it('should handle globalPreload returning undefined', async () => {
-    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
-      '--no-warnings',
-      '--experimental-loader',
-      'data:text/javascript,export function globalPreload(){}',
-      fixtures.path('empty.js'),
-    ]);
+  describe('globalPreload', () => {
+    it('should handle globalPreload returning undefined', async () => {
+      const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+        '--no-warnings',
+        '--experimental-loader',
+        'data:text/javascript,export function globalPreload(){}',
+        fixtures.path('empty.js'),
+      ]);
 
-    assert.strictEqual(stderr, '');
-    assert.strictEqual(stdout, '');
-    assert.strictEqual(code, 0);
-    assert.strictEqual(signal, null);
+      assert.strictEqual(stderr, '');
+      assert.strictEqual(stdout, '');
+      assert.strictEqual(code, 0);
+      assert.strictEqual(signal, null);
+    });
+
+    it('should handle loading node:test', async () => {
+      const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+        '--no-warnings',
+        '--experimental-loader',
+        'data:text/javascript,export function globalPreload(){return `getBuiltin("node:test")()`}',
+        fixtures.path('empty.js'),
+      ]);
+
+      assert.strictEqual(stderr, '');
+      assert.match(stdout, /\n# pass 1\r?\n/);
+      assert.strictEqual(code, 0);
+      assert.strictEqual(signal, null);
+    });
+
+    it('should handle loading node:os with node: prefix', async () => {
+      const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+        '--no-warnings',
+        '--experimental-loader',
+        'data:text/javascript,export function globalPreload(){return `console.log(getBuiltin("node:os").arch())`}',
+        fixtures.path('empty.js'),
+      ]);
+
+      assert.strictEqual(stderr, '');
+      assert.strictEqual(stdout.trim(), os.arch());
+      assert.strictEqual(code, 0);
+      assert.strictEqual(signal, null);
+    });
+
+    it('should handle loading node:os without node: prefix', async () => {
```

Comment:
I _think_ the builtin module used doesn't particularly matter (`os` is merely chosen because it's simple and others like `fs` get mocked under some circumstances)? If so, I think the case description for this and the case above it should reflect that

```suggestion
    // `os` is used here because it's simple and not mocked (the builtin module otherwise doesn't matter).
    it('should handle loading builtin module without node: prefix', async () => {
```